modules/persistence/src/services/metadata/repos_branches/index.ts

```diff
@@ -16,17 +16,23 @@ export interface BranchEntry {
   [key: string]: any;
 }
 
+export interface DocsetsDocument extends WithId<Document> {
+  url: EnvKeyedObject;
+  prefix: EnvKeyedObject;
+  bucket: EnvKeyedObject;
+}
+
 export interface ReposBranchesDocument extends WithId<Document> {
   repoName: string;
   project: string;
   branches: BranchEntry[];
-  url: EnvKeyedObject;
-  prefix: EnvKeyedObject;
   internalOnly: boolean;
   [key: string]: any;
 }
 
-const internals: { [key: project]: ReposBranchesDocument } = {};
+export type ReposBranchesDocsetsDocument = ReposBranchesDocument & DocsetsDocument;
```

Comment:
what is this new document type?

IIUC it contains `url, prefix, bucket` but also `repoName, project, ...`

Reply:
That is correct. The new document type is a combination of `repos_branches` and `docsets`. The following properties come from `docsets` now, which are `prefix`, `bucket` and `URL`.

Reply:
are we constructing a document with both of these properties? seeing that neither `repos_branches` or `docsets` has all of these properties defined. think we should be defining `ReposBranchesDocument` and `DocsetsDocument` separately, not combining them into one. (if connecting the two types, it should be that docsets have a `repos` property that is `ReposBranchesDocument[]` as in the data:

![image](https://github.com/mongodb/docs-worker-pool/assets/13054820/ded0df3b-64ce-437b-876c-7a1ac7011105)

![image](https://github.com/mongodb/docs-worker-pool/assets/13054820/07d5b8bf-e86a-4e1a-8630-2bb34bc4da4c)

Reply:
We're aggregating the two together to have a flat access to all of the properties!